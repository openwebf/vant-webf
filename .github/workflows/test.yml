name: CI

on:
  push:
    branches:
      - '**'

  pull_request:
    branches:
      - dev

  workflow_dispatch:

jobs:
  lint:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run linter
      run: npm run lint

  test:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Run test cases
      run: npm test

    - name: Upload coverage to Codecov  
      uses: codecov/codecov-action@v1
      with:
        token: ${{ secrets.CODECOV_TOKEN }}

  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2

    - name: Install pnpm
      run: npm i pnpm@7 -g

    - uses: actions/setup-node@v2
      with:
        node-version: '16'
        cache: 'pnpm'

    - name: Install dependencies
      run: pnpm install

    - name: Build
      run: npm run build
