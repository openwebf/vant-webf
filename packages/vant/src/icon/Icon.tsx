import {
  computed,
  defineComponent,
  type ExtractPropTypes,
  inject,
  type PropType,
} from 'vue';
import {
  addUnit,
  createNamespace,
  makeStringProp,
  numericProp,
} from '../utils';
import { Badge, type BadgeProps } from '../badge';
import { CONFIG_PROVIDER_KEY } from '../config-provider/ConfigProvider';
import codeMap from './config';

const [name, bem] = createNamespace('icon');

const isImage = (name?: string) => name?.includes('/');

const iconProps = {
  dot: Boolean,
  tag: makeStringProp<keyof HTMLElementTagNameMap>('i'),
  name: String,
  size: numericProp,
  badge: numericProp,
  color: String,
  badgeProps: Object as PropType<Partial<BadgeProps>>,
  classPrefix: String,
};

export type IconProps = ExtractPropTypes<typeof iconProps>;

export default defineComponent({
  name,

  props: iconProps,

  setup(props) {
    const config = inject(CONFIG_PROVIDER_KEY, null);

    const classPrefix = computed(
      () => props.classPrefix || config?.iconPrefix || bem()
    );
    return () => {
      const { tag, dot, name, size, badge, color } = props;
      const isImageIcon = isImage(name);
      const renderCode = (iconName: any) => (
        <span innerHTML={`&#x${codeMap[iconName].content}`}></span>
      );
      return (
        <Badge
          dot={dot}
          tag={tag}
          class={[
            classPrefix.value,
            isImageIcon ? '' : `${classPrefix.value}-${name}`,
          ]}
          style={{
            color,
            fontSize: addUnit(size),
          }}
          content={badge}
          {...props.badgeProps}
        >
          {isImageIcon ? (
            <img class={bem('image')} src={name} />
          ) : (
            renderCode(`${classPrefix.value}-${name}`)
          )}
        </Badge>
      );
    };
  },
});
