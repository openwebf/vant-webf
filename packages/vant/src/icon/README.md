# Icon

### Intro

The font-based icon set that can be used via the Icon component or referenced in other components via the `icon` attribute.

### Install

Register component globally via `app.use`, refer to [Component Registration](#/en-US/advanced-usage#zu-jian-zhu-ce) for more registration ways.

```js
import { createApp } from 'vue';
import { Icon } from 'vant';

const app = createApp();
app.use(Icon);
```

## Usage

### Basic Usage

Use `name` prop to set icon name or icon URL.

```html
<van-icon name="chat-o" />
```

### Using URL

You can directly passing an image URL to the `name` props.

```html
<van-icon name="https://fastly.jsdelivr.net/npm/@vant/assets/icon-demo.png" />
```

### Show Badge

Use `dot` prop, a small red dot will be displayed in the upper right corner of the icon.

Use `badge` prop, the badge will be displayed in the upper right corner of the icon.

```html
<van-icon name="chat-o" dot />
<van-icon name="chat-o" badge="9" />
<van-icon name="chat-o" badge="99+" />
```

### Icon Color

Use `color` prop to set icon color.

```html
<van-icon name="cart-o" color="#1989fa" />
<van-icon name="fire-o" color="#ee0a24" />
```

### Icon Size

Use `size` prop to set icon size.

```html
<!-- Using px unit by default -->
<van-icon name="chat-o" size="40" />
<!-- Using rem unit -->
<van-icon name="chat-o" size="3rem" />
```

### Add custom iconfont

```css
@font-face {
  font-family: 'my-icon';
  src: url('./my-icon.ttf') format('truetype');
}

.my-icon {
  font-family: 'my-icon';
}

.my-icon-extra::before {
  content: '\e626';
}
```

```html
<van-icon class-prefix="my-icon" name="extra" />
```

## API

### Props

| Attribute | Description | Type | Default |
| --- | --- | --- | --- |
| name | Icon name or URL | _string_ | `''` |
| dot | Whether to show red dot | _boolean_ | `false` |
| badge | Content of the badge | _number \| string_ | `''` |
| badge-props `v3.2.8` | Props of Badge, see [Badge - props](#/en-US/badge#props) | _BadgeProps_ | - |
| color | Icon color | _string_ | `inherit` |
| size | Icon size | _number \| string_ | `inherit` |
| class-prefix | ClassName prefix | _string_ | `van-icon` |
| tag | HTML Tag of root element | _string_ | `i` |

### Events

| Event | Description                  | Arguments           |
| ----- | ---------------------------- | ------------------- |
| click | Emitted when icon is clicked | _event: MouseEvent_ |

### Types

The component exports the following type definitions:

```ts
import type { IconProps } from 'vant';
```
