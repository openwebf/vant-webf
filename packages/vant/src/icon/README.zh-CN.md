# Icon 图标

### 介绍

基于字体的图标集，可以通过 Icon 组件使用，也可以在其他组件中通过 `icon` 属性引用。

### 引入

通过以下方式来全局注册组件，更多注册方式请参考[组件注册](#/zh-CN/advanced-usage#zu-jian-zhu-ce)。

```js
import { createApp } from 'vue';
import { Icon } from 'vant';

const app = createApp();
app.use(Icon);
```

## 代码演示

### 基础用法

通过 `name` 属性来指定需要使用的图标，Vant 内置了一套图标库（见右侧示例），可以直接传入对应的名称来使用。

```html
<van-icon name="chat-o" />
```

### 使用图片 URL

你也可以直接在 `name` 属性中传入一个图片 URL 来作为图标。

```html
<van-icon name="https://fastly.jsdelivr.net/npm/@vant/assets/icon-demo.png" />
```

### 徽标提示

设置 `dot` 属性后，会在图标右上角展示一个小红点；设置 `badge` 属性后，会在图标右上角展示相应的徽标。

```html
<van-icon name="chat-o" dot />
<van-icon name="chat-o" badge="9" />
<van-icon name="chat-o" badge="99+" />
```

### 图标颜色

通过 `color` 属性来设置图标的颜色。

```html
<van-icon name="cart-o" color="#1989fa" />
<van-icon name="fire-o" color="#ee0a24" />
```

### 图标大小

通过 `size` 属性来设置图标的尺寸大小，可以指定任意 CSS 单位。

```html
<!-- 不指定单位，默认使用 px -->
<van-icon name="chat-o" size="40" />
<!-- 指定使用 rem 单位 -->
<van-icon name="chat-o" size="3rem" />
```

### 自定义图标

如果需要在现有 Icon 的基础上使用更多图标，可以引入第三方 iconfont 对应的字体文件和 CSS 文件，之后就可以在 Icon 组件中直接使用。

```css
/* 引入第三方或自定义的字体图标样式 */
@font-face {
  font-family: 'my-icon';
  src: url('./my-icon.ttf') format('truetype');
}

.my-icon {
  font-family: 'my-icon';
}

.my-icon-extra::before {
  content: '\e626';
}
```

```html
<!-- 通过 class-prefix 指定类名为 my-icon -->
<van-icon class-prefix="my-icon" name="extra" />
```

## API

### Props

| 参数 | 说明 | 类型 | 默认值 |
| --- | --- | --- | --- |
| name | 图标名称或图片链接 | _string_ | - |
| dot | 是否显示图标右上角小红点 | _boolean_ | `false` |
| badge | 图标右上角徽标的内容 | _number \| string_ | - |
| badge-props `v3.2.8` | 自定义徽标的属性，传入的对象会被透传给 [Badge 组件的 props](#/zh-CN/badge#props) | _BadgeProps_ | - |
| color | 图标颜色 | _string_ | `inherit` |
| size | 图标大小，如 `20px` `2em`，默认单位为 `px` | _number \| string_ | `inherit` |
| class-prefix | 类名前缀，用于使用自定义图标 | _string_ | `van-icon` |
| tag | 根节点对应的 HTML 标签名 | _string_ | `i` |

### Events

| 事件名 | 说明           | 回调参数            |
| ------ | -------------- | ------------------- |
| click  | 点击图标时触发 | _event: MouseEvent_ |

### 类型定义

组件导出以下类型定义：

```ts
import type { IconProps } from 'vant';
```
