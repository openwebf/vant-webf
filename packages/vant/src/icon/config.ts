type json = {
  [key: string]: any;
};
const config: json = {
  'van-icon-exchange': {
    content: 'e6af',
  },

  'van-icon-eye': {
    content: 'e6b0',
  },

  'van-icon-enlarge': {
    content: 'e6b1',
  },

  'van-icon-expand-o': {
    content: 'e6b2',
  },

  'van-icon-eye-o': {
    content: 'e6b3',
  },

  'van-icon-expand': {
    content: 'e6b4',
  },

  'van-icon-filter-o': {
    content: 'e6b5',
  },

  'van-icon-fire': {
    content: 'e6b6',
  },

  'van-icon-fail': {
    content: 'e6b7',
  },

  'van-icon-failure': {
    content: 'e6b8',
  },

  'van-icon-fire-o': {
    content: 'e6b9',
  },

  'van-icon-flag-o': {
    content: 'e6ba',
  },

  'van-icon-font': {
    content: 'e6bb',
  },

  'van-icon-font-o': {
    content: 'e6bc',
  },

  'van-icon-gem-o': {
    content: 'e6bd',
  },

  'van-icon-flower-o': {
    content: 'e6be',
  },

  'van-icon-gem': {
    content: 'e6bf',
  },

  'van-icon-gift-card': {
    content: 'e6c0',
  },

  'van-icon-friends': {
    content: 'e6c1',
  },

  'van-icon-friends-o': {
    content: 'e6c2',
  },

  'van-icon-gold-coin': {
    content: 'e6c3',
  },

  'van-icon-gold-coin-o': {
    content: 'e6c4',
  },

  'van-icon-good-job-o': {
    content: 'e6c5',
  },

  'van-icon-gift': {
    content: 'e6c6',
  },

  'van-icon-gift-o': {
    content: 'e6c7',
  },

  'van-icon-gift-card-o': {
    content: 'e6c8',
  },

  'van-icon-good-job': {
    content: 'e6c9',
  },

  'van-icon-home-o': {
    content: 'e6ca',
  },

  'van-icon-goods-collect': {
    content: 'e6cb',
  },

  'van-icon-graphic': {
    content: 'e6cc',
  },

  'van-icon-goods-collect-o': {
    content: 'e6cd',
  },

  'van-icon-hot-o': {
    content: 'e6ce',
  },

  'van-icon-info': {
    content: 'e6cf',
  },

  'van-icon-hotel-o': {
    content: 'e6d0',
  },

  'van-icon-info-o': {
    content: 'e6d1',
  },

  'van-icon-hot-sale-o': {
    content: 'e6d2',
  },

  'van-icon-hot': {
    content: 'e6d3',
  },

  'van-icon-like': {
    content: 'e6d4',
  },

  'van-icon-idcard': {
    content: 'e6d5',
  },

  'van-icon-invitation': {
    content: 'e6d6',
  },

  'van-icon-like-o': {
    content: 'e6d7',
  },

  'van-icon-hot-sale': {
    content: 'e6d8',
  },

  'van-icon-location-o': {
    content: 'e6d9',
  },

  'van-icon-location': {
    content: 'e6da',
  },

  'van-icon-label': {
    content: 'e6db',
  },

  'van-icon-lock': {
    content: 'e6dc',
  },

  'van-icon-label-o': {
    content: 'e6dd',
  },

  'van-icon-map-marked': {
    content: 'e6de',
  },

  'van-icon-logistics': {
    content: 'e6df',
  },

  'van-icon-manager': {
    content: 'e6e0',
  },

  'van-icon-more': {
    content: 'e6e1',
  },

  'van-icon-live': {
    content: 'e6e2',
  },

  'van-icon-manager-o': {
    content: 'e6e3',
  },

  'van-icon-medal': {
    content: 'e6e4',
  },

  'van-icon-more-o': {
    content: 'e6e5',
  },

  'van-icon-music-o': {
    content: 'e6e6',
  },

  'van-icon-music': {
    content: 'e6e7',
  },

  'van-icon-new-arrival-o': {
    content: 'e6e8',
  },

  'van-icon-medal-o': {
    content: 'e6e9',
  },

  'van-icon-new-o': {
    content: 'e6ea',
  },

  'van-icon-free-postage': {
    content: 'e6eb',
  },

  'van-icon-newspaper-o': {
    content: 'e6ec',
  },

  'van-icon-new-arrival': {
    content: 'e6ed',
  },

  'van-icon-minus': {
    content: 'e6ee',
  },

  'van-icon-orders-o': {
    content: 'e6ef',
  },

  'van-icon-new': {
    content: 'e6f0',
  },

  'van-icon-paid': {
    content: 'e6f1',
  },

  'van-icon-notes-o': {
    content: 'e6f2',
  },

  'van-icon-other-pay': {
    content: 'e6f3',
  },

  'van-icon-pause-circle': {
    content: 'e6f4',
  },

  'van-icon-pause': {
    content: 'e6f5',
  },

  'van-icon-pause-circle-o': {
    content: 'e6f6',
  },

  'van-icon-peer-pay': {
    content: 'e6f7',
  },

  'van-icon-pending-payment': {
    content: 'e6f8',
  },

  'van-icon-passed': {
    content: 'e6f9',
  },

  'van-icon-plus': {
    content: 'e6fa',
  },

  'van-icon-phone-circle-o': {
    content: 'e6fb',
  },

  'van-icon-phone-o': {
    content: 'e6fc',
  },

  'van-icon-printer': {
    content: 'e6fd',
  },

  'van-icon-photo-fail': {
    content: 'e6fe',
  },

  'van-icon-phone': {
    content: 'e6ff',
  },

  'van-icon-photo-o': {
    content: 'e700',
  },

  'van-icon-play-circle': {
    content: 'e701',
  },

  'van-icon-play': {
    content: 'e702',
  },

  'van-icon-phone-circle': {
    content: 'e703',
  },

  'van-icon-point-gift-o': {
    content: 'e704',
  },

  'van-icon-point-gift': {
    content: 'e705',
  },

  'van-icon-play-circle-o': {
    content: 'e706',
  },

  'van-icon-shrink': {
    content: 'e707',
  },

  'van-icon-photo': {
    content: 'e708',
  },

  'van-icon-qr': {
    content: 'e709',
  },

  'van-icon-qr-invalid': {
    content: 'e70a',
  },

  'van-icon-question-o': {
    content: 'e70b',
  },

  'van-icon-revoke': {
    content: 'e70c',
  },

  'van-icon-replay': {
    content: 'e70d',
  },

  'van-icon-service': {
    content: 'e70e',
  },

  'van-icon-question': {
    content: 'e70f',
  },

  'van-icon-search': {
    content: 'e710',
  },

  'van-icon-refund-o': {
    content: 'e711',
  },

  'van-icon-service-o': {
    content: 'e712',
  },

  'van-icon-scan': {
    content: 'e713',
  },

  'van-icon-share': {
    content: 'e714',
  },

  'van-icon-send-gift-o': {
    content: 'e715',
  },

  'van-icon-share-o': {
    content: 'e716',
  },

  'van-icon-setting': {
    content: 'e717',
  },

  'van-icon-points': {
    content: 'e718',
  },

  'van-icon-photograph': {
    content: 'e719',
  },

  'van-icon-shop': {
    content: 'e71a',
  },

  'van-icon-shop-o': {
    content: 'e71b',
  },

  'van-icon-shop-collect-o': {
    content: 'e71c',
  },

  'van-icon-shop-collect': {
    content: 'e71d',
  },

  'van-icon-smile': {
    content: 'e71e',
  },

  'van-icon-shopping-cart-o': {
    content: 'e71f',
  },

  'van-icon-sign': {
    content: 'e720',
  },

  'van-icon-sort': {
    content: 'e721',
  },

  'van-icon-star-o': {
    content: 'e722',
  },

  'van-icon-smile-comment-o': {
    content: 'e723',
  },

  'van-icon-stop': {
    content: 'e724',
  },

  'van-icon-stop-circle-o': {
    content: 'e725',
  },

  'van-icon-smile-o': {
    content: 'e726',
  },

  'van-icon-star': {
    content: 'e727',
  },

  'van-icon-success': {
    content: 'e728',
  },

  'van-icon-stop-circle': {
    content: 'e729',
  },

  'van-icon-records': {
    content: 'e72a',
  },

  'van-icon-shopping-cart': {
    content: 'e72b',
  },

  'van-icon-tosend': {
    content: 'e72c',
  },

  'van-icon-todo-list': {
    content: 'e72d',
  },

  'van-icon-thumb-circle-o': {
    content: 'e72e',
  },

  'van-icon-thumb-circle': {
    content: 'e72f',
  },

  'van-icon-umbrella-circle': {
    content: 'e730',
  },

  'van-icon-underway': {
    content: 'e731',
  },

  'van-icon-upgrade': {
    content: 'e732',
  },

  'van-icon-todo-list-o': {
    content: 'e733',
  },

  'van-icon-tv-o': {
    content: 'e734',
  },

  'van-icon-underway-o': {
    content: 'e735',
  },

  'van-icon-user-o': {
    content: 'e736',
  },

  'van-icon-vip-card-o': {
    content: 'e737',
  },

  'van-icon-vip-card': {
    content: 'e738',
  },

  'van-icon-send-gift': {
    content: 'e739',
  },

  'van-icon-wap-home': {
    content: 'e73a',
  },

  'van-icon-wap-nav': {
    content: 'e73b',
  },

  'van-icon-volume-o': {
    content: 'e73c',
  },

  'van-icon-video': {
    content: 'e73d',
  },

  'van-icon-wap-home-o': {
    content: 'e73e',
  },

  'van-icon-volume': {
    content: 'e73f',
  },

  'van-icon-warning': {
    content: 'e740',
  },

  'van-icon-weapp-nav': {
    content: 'e741',
  },

  'van-icon-wechat-pay': {
    content: 'e742',
  },

  'van-icon-warning-o': {
    content: 'e743',
  },

  'van-icon-wechat': {
    content: 'e744',
  },

  'van-icon-setting-o': {
    content: 'e745',
  },

  'van-icon-youzan-shield': {
    content: 'e746',
  },

  'van-icon-warn-o': {
    content: 'e747',
  },

  'van-icon-smile-comment': {
    content: 'e748',
  },

  'van-icon-user-circle-o': {
    content: 'e749',
  },

  'van-icon-video-o': {
    content: 'e74a',
  },

  'van-icon-add-square': {
    content: 'e65c',
  },

  'van-icon-add': {
    content: 'e65d',
  },

  'van-icon-arrow-down': {
    content: 'e65e',
  },

  'van-icon-arrow-up': {
    content: 'e65f',
  },

  'van-icon-arrow': {
    content: 'e660',
  },

  'van-icon-after-sale': {
    content: 'e661',
  },

  'van-icon-add-o': {
    content: 'e662',
  },

  'van-icon-alipay': {
    content: 'e663',
  },

  'van-icon-ascending': {
    content: 'e664',
  },

  'van-icon-apps-o': {
    content: 'e665',
  },

  'van-icon-aim': {
    content: 'e666',
  },

  'van-icon-award': {
    content: 'e667',
  },

  'van-icon-arrow-left': {
    content: 'e668',
  },

  'van-icon-award-o': {
    content: 'e669',
  },

  'van-icon-audio': {
    content: 'e66a',
  },

  'van-icon-bag-o': {
    content: 'e66b',
  },

  'van-icon-balance-list': {
    content: 'e66c',
  },

  'van-icon-back-top': {
    content: 'e66d',
  },

  'van-icon-bag': {
    content: 'e66e',
  },

  'van-icon-balance-pay': {
    content: 'e66f',
  },

  'van-icon-balance-o': {
    content: 'e670',
  },

  'van-icon-bar-chart-o': {
    content: 'e671',
  },

  'van-icon-bars': {
    content: 'e672',
  },

  'van-icon-balance-list-o': {
    content: 'e673',
  },

  'van-icon-birthday-cake-o': {
    content: 'e674',
  },

  'van-icon-bookmark': {
    content: 'e675',
  },

  'van-icon-bill': {
    content: 'e676',
  },

  'van-icon-bell': {
    content: 'e677',
  },

  'van-icon-browsing-history-o': {
    content: 'e678',
  },

  'van-icon-browsing-history': {
    content: 'e679',
  },

  'van-icon-bookmark-o': {
    content: 'e67a',
  },

  'van-icon-bulb-o': {
    content: 'e67b',
  },

  'van-icon-bullhorn-o': {
    content: 'e67c',
  },

  'van-icon-bill-o': {
    content: 'e67d',
  },

  'van-icon-calendar-o': {
    content: 'e67e',
  },

  'van-icon-brush-o': {
    content: 'e67f',
  },

  'van-icon-card': {
    content: 'e680',
  },

  'van-icon-cart-o': {
    content: 'e681',
  },

  'van-icon-cart-circle': {
    content: 'e682',
  },

  'van-icon-cart-circle-o': {
    content: 'e683',
  },

  'van-icon-cart': {
    content: 'e684',
  },

  'van-icon-cash-on-deliver': {
    content: 'e685',
  },

  'van-icon-cash-back-record': {
    content: 'e686',
  },

  'van-icon-cashier-o': {
    content: 'e687',
  },

  'van-icon-chart-trending-o': {
    content: 'e688',
  },

  'van-icon-certificate': {
    content: 'e689',
  },

  'van-icon-chat': {
    content: 'e68a',
  },

  'van-icon-clear': {
    content: 'e68b',
  },

  'van-icon-chat-o': {
    content: 'e68c',
  },

  'van-icon-checked': {
    content: 'e68d',
  },

  'van-icon-clock': {
    content: 'e68e',
  },

  'van-icon-clock-o': {
    content: 'e68f',
  },

  'van-icon-close': {
    content: 'e690',
  },

  'van-icon-closed-eye': {
    content: 'e691',
  },

  'van-icon-circle': {
    content: 'e692',
  },

  'van-icon-cluster-o': {
    content: 'e693',
  },

  'van-icon-column': {
    content: 'e694',
  },

  'van-icon-comment-circle-o': {
    content: 'e695',
  },

  'van-icon-cluster': {
    content: 'e696',
  },

  'van-icon-comment': {
    content: 'e697',
  },

  'van-icon-comment-o': {
    content: 'e698',
  },

  'van-icon-comment-circle': {
    content: 'e699',
  },

  'van-icon-completed': {
    content: 'e69a',
  },

  'van-icon-credit-pay': {
    content: 'e69b',
  },

  'van-icon-coupon': {
    content: 'e69c',
  },

  'van-icon-debit-pay': {
    content: 'e69d',
  },

  'van-icon-coupon-o': {
    content: 'e69e',
  },

  'van-icon-contact': {
    content: 'e69f',
  },

  'van-icon-descending': {
    content: 'e6a0',
  },

  'van-icon-desktop-o': {
    content: 'e6a1',
  },

  'van-icon-diamond-o': {
    content: 'e6a2',
  },

  'van-icon-description': {
    content: 'e6a3',
  },

  'van-icon-delete': {
    content: 'e6a4',
  },

  'van-icon-diamond': {
    content: 'e6a5',
  },

  'van-icon-delete-o': {
    content: 'e6a6',
  },

  'van-icon-cross': {
    content: 'e6a7',
  },

  'van-icon-edit': {
    content: 'e6a8',
  },

  'van-icon-ellipsis': {
    content: 'e6a9',
  },

  'van-icon-down': {
    content: 'e6aa',
  },

  'van-icon-discount': {
    content: 'e6ab',
  },

  'van-icon-ecard-pay': {
    content: 'e6ac',
  },

  'van-icon-envelop-o': {
    content: 'e6ae',
  },

  'van-icon-shield-o': {
    content: 'e74b',
  },

  'van-icon-guide-o': {
    content: 'e74c',
  },

  'van-icon-cash-o': {
    content: 'e74d',
  },

  'van-icon-qq': {
    content: 'e74e',
  },

  'van-icon-wechat-moments': {
    content: 'e74f',
  },

  'van-icon-weibo': {
    content: 'e750',
  },

  'van-icon-link-o': {
    content: 'e751',
  },

  'van-icon-miniprogram-o': {
    content: 'e752',
  },
};
export default config;
