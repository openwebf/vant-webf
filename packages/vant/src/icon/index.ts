import { withInstall } from '../utils';
import _Icon from './Icon';

export const Icon = withInstall(_Icon);
export default Icon;
export type { IconProps } from './Icon';

declare module 'vue' {
  export interface GlobalComponents {
    VanIcon: typeof Icon;
  }
}
