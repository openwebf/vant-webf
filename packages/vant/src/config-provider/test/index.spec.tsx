import { ref } from 'vue';
import { ConfigProvider } from '..';
import { Icon } from '../../icon';
import { later, mount } from '../../../test';
import Popup from '../../popup';

test('should render tag prop correctly', () => {
  const wrapper = mount(ConfigProvider, {
    props: {
      tag: 'section',
    },
  });
  expect(wrapper.html()).toMatchSnapshot();
});

test('should change icon class-prefix when using icon-prefix prop', () => {
  const wrapper = mount({
    render() {
      return (
        <ConfigProvider iconPrefix="foo">
          <Icon name="success" />
        </ConfigProvider>
      );
    },
  });
  expect(wrapper.html()).toMatchSnapshot();
});

test('should change global z-index when using z-index prop', async () => {
  const show = ref(true);
  const wrapper = mount({
    render() {
      return (
        <ConfigProvider zIndex={0}>
          <Popup v-model:show={show.value} />
        </ConfigProvider>
      );
    },
  });

  await later();
  expect(wrapper.find('.van-popup').style.zIndex).toEqual('1');
});
