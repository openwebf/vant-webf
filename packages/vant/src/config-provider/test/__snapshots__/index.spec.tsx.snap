// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should change icon class-prefix when using icon-prefix prop 1`] = `
<div class="van-config-provider">
  <i class="van-badge__wrapper foo foo-success">
  </i>
</div>
`;

exports[`should render tag prop correctly 1`] = `
<section class="van-config-provider">
</section>
`;
